tests/unit/utils/sparqlhandler/test_data_cube_state.py

```diff
@@ -268,14 +281,257 @@ def test_detect_csvw_shape_standard():
+        _test_case_base_dir
+        / "pivoted-multi-measure-dataset"
+        / "qb-id-10003.csv-metadata.json"
+    )
+
+    data_cube_state = get_data_cube_inspector(path_to_json_file)
+
+    result_qube_components: QubeComponentsResult = (
+        data_cube_state.get_dsd_qube_components_for_csv("qb-id-10003.csv")
+    )
+
+    components = result_qube_components.qube_components
+    assert len(components) == 7
+
+    component = get_dsd_component_by_property_url(
+        components, "qb-id-10003.csv#dimension/some-dimension"
+    )
+    assert_dsd_component_equal(
+        component,
+        "qb-id-10003.csv#dimension/some-dimension",
+        ComponentPropertyType.Dimension,
+        "Some Dimension",
+        ["Some Dimension"],
+        ["Some Obs Val", "Some Other Obs Val"],
+        "qb-id-10003.csv#structure",
+        True,
+    )
+
+    component = get_dsd_component_by_property_url(
+        components, "qb-id-10003.csv#attribute/some-attribute"
+    )
+    assert_dsd_component_equal(
+        component,
+        "qb-id-10003.csv#attribute/some-attribute",
+        ComponentPropertyType.Attribute,
+        "Some Attribute",
+        ["Some Attribute"],
+        ["Some Obs Val"],
+        "qb-id-10003.csv#structure",
+        False,
+    )
+
+    component = get_dsd_component_by_property_url(
+        components, "http://purl.org/linked-data/cube#measureType"
+    )
+    assert_dsd_component_equal(
+        component,
+        "http://purl.org/linked-data/cube#measureType",
+        ComponentPropertyType.Dimension,
+        "",
+        [],
+        [],
+        "qb-id-10003.csv#structure",
+        True,
+    )
+
+    component = get_dsd_component_by_property_url(
+        components, "http://purl.org/linked-data/sdmx/2009/attribute#unitMeasure"
+    )
+    assert_dsd_component_equal(
+        component,
+        "http://purl.org/linked-data/sdmx/2009/attribute#unitMeasure",
+        ComponentPropertyType.Attribute,
+        "",
+        ["Some Unit"],
+        ["Some Other Obs Val", "Some Obs Val"],
+        "qb-id-10003.csv#structure",
+        True,
+    )
+
+    component = get_dsd_component_by_property_url(
+        components, "qb-id-10003.csv#measure/some-measure"
+    )
+    assert_dsd_component_equal(
+        component,
+        "qb-id-10003.csv#measure/some-measure",
+        ComponentPropertyType.Measure,
+        "Some Measure",
+        ["Some Obs Val"],
+        ["Some Obs Val"],
+        "qb-id-10003.csv#structure",
+        True,
+    )
+
+    component = get_dsd_component_by_property_url(
+        components, "qb-id-10003.csv#measure/some-other-measure"
+    )
+    assert_dsd_component_equal(
+        component,
+        "qb-id-10003.csv#measure/some-other-measure",
+        ComponentPropertyType.Measure,
+        "Some Other Measure",
+        ["Some Other Obs Val"],
+        ["Some Other Obs Val"],
+        "qb-id-10003.csv#structure",
+        True,
+    )
+
+
+def test_dsd_single_measure_pivoted_shape():
```

Comment:
Do we have any tests for the DSD components of the standard shape?

Reply:
Added a unit test for the DSD components of the standard shape single unit single measure dataset: `/ "single-unit_single-measure" / "energy-trends-uk-total-energy.csv-metadata.json"`